### PR TITLE
fix(llm/lld): refresh user properties when accounts changes

### PR DESCRIPTION
**Pull request by cgrellard-ledger**
### 📝 Description

Refresh user properties when accounts changes

### ❓ Context

- **Impacted projects**: `llm` `lld` 
- **Linked resource(s)**: [LIVE-8837] 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-8837]: https://ledgerhq.atlassian.net/browse/LIVE-8837?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { useEffect, useMemo, useRef } from "react";
+import debounce from "lodash/debounce";
+import type { Account } from "@ledgerhq/types-live";
+
+export function hasAccountsWithFundsChanged(accounts: Account[], oldAccounts: Account[]): boolean {
```
Comment (sshmaxime) [inline libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.ts:5]: Wouldn't it be nice to add some comments/explanation about the function ? Like what do we actually mean by `changed` ?

File: libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.test.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  describe("Test the hasAccountsWithFundsChanged helper function", () => {
+    test("should return true when an account with funds is added", () => {
+      const newAccounts: Account[] = [ethMockAccount];
+      const oldAccounts: Account[] = [];
+
+      const res = hasAccountsWithFundsChanged(newAccounts, oldAccounts);
+
+      expect(res).toBeTruthy();
+    });
+    test("should return true when an account with funds is deleted", () => {
```
Comment (sshmaxime) [inline libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.test.tsx:107]:
```suggestion

    test("should return true when an account with funds is deleted", () => {
```

That's just styling but we should try to stay consistent on that, we should probably all talk about it and agree on a define style sheet to increase consistency in the repo

File: libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        continue;
+      }
+    }
+  }
+  return false;
+}
+
+function useAccountsWithFundsListener(
+  accounts: Account[],
+  cb: () => void,
```
Comment (mcayuelas-ledger) [inline libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.ts:52]: is it possible to rename with something more explicit ?

File: libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    // Means that account has been modified
+    const hasAccountBeenEmptied =
+      account?.balance.isZero() && matchingOldAccount?.balance.isGreaterThan(0);
+    const hasAccountReceivedFunds =
+      account?.balance.isGreaterThan(0) && matchingOldAccount?.balance.isZero();
+
+    if (hasAccountBeenEmptied || hasAccountReceivedFunds) {
+      return true;
+    }
```
Comment (mcayuelas-ledger) [inline libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.ts:32]:
```suggestion
    return (hasAccountBeenEmptied || hasAccountReceivedFunds)
```
Could works :p

File: libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.test.tsx
Comment (mcayuelas-ledger) [inline libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.test.tsx:1]: Nice job 🚀

File: libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.ts (hunk 4 above)
Comment (cgrellard-ledger) [reply libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.ts:32]: If none of the two is true we need to execute the second for loop ;)

**Review by mcayuelas-ledger — APPROVED** [review]

**Pull request merged**